core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -478,10 +579,14 @@ public void run()
                     }
 
                     String threadId = split.getTaskHandle().getTaskId() + "-" + split.getSplitId();
+
                     try (SetThreadName splitName = new SetThreadName(threadId)) {
-                        RunningSplitInfo splitInfo = new RunningSplitInfo(ticker.read(), threadId, Thread.currentThread());
-                        runningSplitInfos.add(splitInfo);
+                        RunningSplitInfo splitInfo = new RunningSplitInfo(ticker.read(), threadId, Thread.currentThread(), split, this);
+                        synchronized (this) {
+                            currentThreadSplitInfo = splitInfo;
```

Comment:
Thanks, that is a good point!

Given this, and https://github.com/trinodb/trino/pull/12479#discussion_r881715903, I am thinking using the taskManager.abortTask(taskId) instead of implementing its own interrupt mechanism.

This can avoid the race condition that the interrupt thread and other threads concurrently trying to abort a task.

How do you think?

Reply:
Yeah, I think it's a good idea. You can try to use either `SqlTask#failed` or `SqlTaskManager#failTask`. These methods will allow you to provide an exception to be used to fail task with where you can include a meaningful error message.